### ISSUE TITLE
Avoid duplicate attachments

### ISSUE THREAD

**Issue by vjfernan** [globaleaks/GlobaLeaks#3749]
### Proposal

When a whistleblower attaches evidence he can attach the same file several times:

![image](https://github.com/globaleaks/GlobaLeaks/assets/23368434/2abbb59c-bc3e-406f-80a5-819a1a6601a9)

The application could check on the client side that an attachment has already been included and display a warning message informing about it.

### Motivation and context

In the storage system duplicate files are stored as separate files taking up more space than necessary.

Whistleblowers and recipients see repeated documents in the attachment list:

![image](https://github.com/globaleaks/GlobaLeaks/assets/23368434/8fc939de-d8f2-4816-a38c-99367ebb6cd4)

**Comment by evilaliv3** [comment]
Thank you @vjfernan 

Do you think that checking name and size and discard the file if already present in the list of uploaded files would be enough?

Ensuring the file is exactly the same is not easy. One could calculate the hash but this is not easy on the client with the capabilities offered by browsers because it is difficult to achieve good performances expecially with large files and the operation should be performed in streaming that is not so easy.

\cc @gianlucagilardi @lorenzogentile @giorgio: given your use cases would you find this useful for a general implmentation within the software?

**Comment by giorgiofraschini** [comment]
I think the software should only stop the attachment of a file with the same name. this would be definitely enough.

**Comment by evilaliv3** [comment]
Ok thank you @giorgiofraschini

I think we could block this and check the length to ensure they are quite for sure the same file.

**Comment by vjfernan** [comment]
Yes, I think checking the name (with its extension) and the length of the file on the client side might be enough.

**Comment by evilaliv3** [comment]
We could definitely implement this preventing:
- upload of files with the same name
- upload of files with the same hash
